### PR TITLE
datasetRw: upload bug fixes

### PR DESCRIPTION
**Pull request by ddl-giuliocapolino**
### Link to JIRA

https://dominodatalab.atlassian.net/browse/DOM-53774
https://dominodatalab.atlassian.net/browse/DOM-53775
https://dominodatalab.atlassian.net/browse/DOM-53779
https://dominodatalab.atlassian.net/browse/DOM-53780

### What issue does this pull request solve?

Take care of them

### What is the solution?

_placeholder_

### Testing

Briefly describe how the change was tested. The purpose of this section is that a reviewer can identify a test gap, if any.

_e.g. "I ran an upgrade from 4.2 to 4.6"._

- [ ] Unit test(s)

### Pull Request Reminders

- [ ] Has relevant documentation been updated?
- [ ] Does the code follow [Python Style Guide] (https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html)
- [ ] Update the [changelog](https://github.com/dominodatalab/python-domino/blob/master/CHANGELOG.md)
- [ ] Are the existing unit tests still passing?
- [ ] Have new unit tests been added to cover any changes to the code?
- [x] Has the JIRA ticket(s) been linked above?

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -14,6 +14,8 @@ All notable changes to the `python-domino` library will be documented in this fi
 ### Added
 * Updated Apps documentation
 * Added get_blobs_v2 endpoint plus tests
+* Added datasets_upload_files endpoint plus tests
+* Updated Projects and Datasets documentation
```
Comment (ddl-olsonJD) [inline CHANGELOG.md:18]: move to unreleased.

Comment (ddl-olsonJD) [reply CHANGELOG.md:18]: Also add tests

**Review by ddl-olsonJD — CHANGES_REQUESTED** [review]

File: domino/domino.py [hunk 2]
```diff
@@ -1054,21 +1054,49 @@ def datasets_upload_files(
...
             max_workers: max amount of threads (default: 10)
             target_chunk_size: max chunk size for multipart upload (default: 8MB)
             target_relative_path: path in the dataset to upload the file or directory to
         Returns local path to uploaded file
         """
+        # interactive part to warn user about setting selected
+        if interactive:
+            user_selection = None
+            while user_selection != "Y":
+                if not file_upload_setting or file_upload_setting == "Overwrite":
```
Comment (ddl-gabrielhaim) [inline domino/domino.py:1081]: nit: check for `is None` because empty string evaluate to False

**Review by ddl-gabrielhaim — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (ddl-giuliocapolino) [reply CHANGELOG.md]: ah no need to add these then. They are already under “Unreleased” from yesterday’s PR. Added tests!

**Review by ddl-olsonJD — APPROVED** [review]

**Pull request merged**